envoy/upstream/upstream.h

```diff
@@ -1060,9 +1060,9 @@ class ClusterInfo : public Http::FilterChainFactory {
   virtual ClusterEndpointStats& endpointStats() const PURE;
 
   /**
-   * @return ClusterTrafficStats& all traffic related stats for this cluster.
+   * @return std::unique_ptr<ClusterTrafficStats>&  all traffic related stats for this cluster.
```

Comment:
done

Reply:
Since we're not passing the ownership here, just use raw pointer `ClusterTrafficStats*` instead of returning a reference to unique ptr?

Reply:
This is a good point, Lizan. However that will not be convenient for the follow-up for this PR.

That's because we want to (once we sort out some semantic issues around stat continuity) return a LazyInit<ClusterTrafficCtats>& here.

I had suggested making this change as an intermediate step because it is a large PR but is (arguably) obviously correct.

Then when we have solved stat continuity across xds updates we would be able to pass around the lazy-init reference, and then the first time a stat is incremented, we'll instantiate the block of stats.

I think this might appear more palatable once Xin adds a nickname for this.

Reply:
Sounds good, if this is an intermediate state just add some comment explaining. A nickname helps as well.